### PR TITLE
User

### PR DESCRIPTION
**Pull request by PranavBakre**
### TODO:
- [X] 1. Add admin routes for updating user and deleting user
- [x] 2. To implement is_active and admin approval
- [X] 3. To add expiry to access token and think of a logic for token refresh
~~- [ ] 4. To implement a class CRUDBase for CRUD operations( need to think if this is a good idea) to use instead of MongoEngine directly~~
- [x] 5. Response To remove or not to remove ??

### CODE REVIEW

**Comment by prkprime** [comment]
why double quotes??????????????????????

**Comment by PranavBakre** [comment]
@prkprime I did say i prefer double quotes

**Comment by prkprime** [comment]
> @prkprime I did say i prefer double quotes

BC

**Comment by PranavBakre** [comment]
@prkprime Dont curse on public repos

**Comment by prkprime** [comment]
> @prkprime Dont curse on public repos

we hab 69 and 420 in our code xD
noone cares

**Review by vigu184 — APPROVED** [review]

File: app/api/api_v1/endpoints/user.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    if update.email is not None:
+        queried_user.email = update.email
+    if update.phone is not None and is_phone_valid(update.phone):
+        queried_user.phone = update.phone
+    if update.password is not None:
+        queried_user.password = get_password_hash(update.password)
+    if update.is_admin is not None:
+        queried_user.is_admin = update.is_admin
+    queried_user.save()
+    return queried_user
```
Comment (prkprime) [inline app/api/api_v1/endpoints/user.py:83]: this function allows admin to update normal user right?

File: app/api/api_v1/endpoints/user.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+@router.delete("/{id}", response_model=UserSchema)
+def delete_user_by_id(user:User = Depends(get_current_admin),
+    id: str = Path(...)
+) -> User:
+    if not ObjectId.is_valid(id):
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail="Invalid Id")
+    queried_user = User.objects(id=id).first()
+    queried_user.delete()
+    return queried_user
```
Comment (prkprime) [inline app/api/api_v1/endpoints/user.py]: admin can delete another admin
is that intended?

**Review by prkprime — COMMENTED** [review]
so far everything looks fine
also @PranavBakre can write better python than me
me sed :(

File: app/api/api_v1/endpoints/user.py (hunk 2 above)
Comment (vigu184) [reply app/api/api_v1/endpoints/user.py]: Not needed. As we decide admin!

**Comment by rittmang** [comment]
Weeew

**Comment by PranavBakre** [comment]
will do CRUDBase in new branch

**Comment by rittmang** [comment]
@PranavBakre Go see Croods

**Review by rittmang — APPROVED** [review]
Bugre Ultra Pro Max

**Comment by PranavBakre** [comment]
> Bugre Ultra Pro Max

rottmang

**Review by vigu184 — APPROVED** [review]

**Review by prkprime — APPROVED** [review]
noice

**Pull request merged**